lib_comfyui/queue_tracker.py

```diff
@@ -0,0 +1,100 @@
+import multiprocessing
+from lib_comfyui import ipc
+
+
+class PromptQueueTracker:
+    done_event = multiprocessing.Event()
+    tracked_id = None
+    queue_instance = None
+    server_instance = None
+
+    @ipc.confine_to('comfyui')
+    @staticmethod
+    def wait_for_last_put():
+        if not PromptQueueTracker.tracked_id_present():
+            return
+
+        PromptQueueTracker.done_event.wait()
+        PromptQueueTracker.done_event.clear()
+
+    @ipc.confine_to('comfyui')
+    @staticmethod
+    def tracked_id_present():
+        with PromptQueueTracker.queue_instance.mutex:
+            for v in PromptQueueTracker.queue_instance.currently_running.values():
+                if abs(v[0]) == PromptQueueTracker.tracked_id:
+                    return True
+            for x in PromptQueueTracker.queue_instance.queue:
+                if abs(x[0]) == PromptQueueTracker.tracked_id:
+                    return True
+            return False
+
+
+    @ipc.confine_to('comfyui')
+    @staticmethod
+    def update_tracked_id():
+        PromptQueueTracker.tracked_id = PromptQueueTracker.server_instance.number
+
+    @staticmethod
+    def patched__init__(self, server_instance):
+        prompt_queue = self
+        PromptQueueTracker.server_instance = server_instance
+        PromptQueueTracker.queue_instance = self
+
+        # task_done
+        original_task_done = prompt_queue.task_done
+
+        def patched_task_done(item_id, output, *args, **kwargs):
+            with prompt_queue.mutex:
+                v = prompt_queue.currently_running[item_id]
+                if abs(v[0]) == PromptQueueTracker.tracked_id:
+                    PromptQueueTracker.done_event.set()
+
+            return original_task_done(item_id, output, *args, **kwargs)
+
+        prompt_queue.task_done = patched_task_done

```

Comment:
What do you think of

```suggestion
        prompt_queue.task_done = functools.partial(patched_task_done, original_function=prompt_queue.task_done)

```

It makes it possible to remove the line above `original_task_done = prompt_queue.task_done` and keep the original function in scope.

Your call, I'm not fighting for this one either.